core/src/main/java/com/microsoft/applicationinsights/common/SanitizationUtils.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.applicationinsights.common;
+
+import java.text.StringCharacterIterator;
+
+/**
+ * Created by dhdoshi on 09/10/2017
+ *
+ * This class provides utility functions to sanitize strings
+ * for various formats. Currently it supports JSON sanitization
+ */
+public final class SanitizationUtils {
+
+    /**
+     * This method appends escape characters to input String to prevent
+     * JSON Sanitization failure
+     * @param text
+     * @return Sanitized String suitable for JSON
+     */
+    public static String sanitizeStringForJSON(String text) {
```

Comment:
I suggest to pass the max length as an additional argument so you do not need to sanitize excessively long string

Reply:
@SergeyKanzhelev I addressed this in my current commits. Please have a look. This solves #403 also.

Reply:
@SergeyKanzhelev Probably I found a potential issue with this. Lets think of a situation when we get an excessively long string almost around max length which has large number of escape characters which will need to be sanitized. Now we truncate it before sanitization but post sanitization the string will again be larger so breeze will respond with 500 if we send that. Quickfix: Truncate before and after sanitization. 
Corner Case: post sanitize truncate can remove some of the sanitized characters and cause the same problem.
Concrete Solution: before sanitization count the occurnace of the characters to be handeled and truncate to that length so sanitiaztion does not exceed max length.
Please let me know your opinion!